File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int l=0,r=height.size()-1,lmax=height[l],rmax=height[r];
        int ans=0;
        while(l<r){
            if(height[l]<height[r]){
                lmax=max(lmax,height[l]);
                ans+=lmax-height[l];
                l++;
            }
            else{
                rmax=max(rmax,height[r]);
                ans+=rmax-height[r];
                r--;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
       int n=nums.length;
       int []ans=new int[n*2];
       for(int i=0;i<2*n;i++){
        ans[i]=nums[i%n];
       } 
       return ans;
    }
}